src/Plugin/rest/resource/GetUserPreferences.php

```diff
@@ -0,0 +1,190 @@
+<?php
+namespace Drupal\ccf_user_preferences\Plugin\rest\resource;
+
+use Drupal\rest\Plugin\ResourceBase;
+use Drupal\user\Entity\User;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\rest\ModifiedResourceResponse;
+use Drupal\jwt\Transcoder\JwtTranscoderInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Drupal\ccf_user_preferences\Entity\CcfUserPreferences;
+
+/**
+ * Provides REST API for User preferences Details based
+ * on URL.
+ *
+ * @RestResource(
+ *   id = "get_user_preferences_rest_resource",
+ *   label = @Translation("Get User Preferences"),
+ *   uri_paths = {
+ *     "canonical" = "/api/get_user_preferences",
+ *     "create"    = "/api/add_user_preferences"
```

Comment:
Can you please check if the base path of the APIs can be managed with the REST UI module? If we can manage the base path of APIs something like `/api/v1` with the REST UI module then here we can only add `/get_user_preferences` and `/add_user_preferences`. 

If we can't manage the base path of APIs with REST UI module then please change base path to `/api/v1` .